Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/Internal/CoreDocumentSynchronizationConfig.swift

```diff
@@ -86,10 +89,15 @@ final class CoreDocumentSynchronization: Codable, Hashable {
     var isStale: Bool {
         get {
             return docLock.read {
-                var filter = docConfigFilter(forNamespace: namespace, withDocumentId: documentId.bsonValue)
+                var filter = docConfigFilter(forNamespace: namespace, withDocumentId: documentId).map {
```

Comment:
[task] can you add a quick comment explaining what's happening here? it seems like a contrived way of adding a field to a BSON document that I assume is because of a Swift driver change, but we should be clear about it